Editor/Editors/CinemachineVirtualCameraEditor.cs

```diff
@@ -390,9 +443,21 @@ void UpdateComponentEditors()
                 m_componentEditors = new UnityEditor.Editor[numComponents];
                 for (int i = 0; i < numComponents; ++i)
                 {
-                    MonoBehaviour b = components[i] as MonoBehaviour;
-                    if (b != null)
-                        CreateCachedEditor(b, null, ref m_componentEditors[i]);
+                    List<MonoBehaviour> behaviours = new List<MonoBehaviour>();
+                    for (int j = 0; j < targets.Length; j++)
+                    {
+                        var cinemachineVirtualCamera = targets[j] as CinemachineVirtualCamera;
+                        if (cinemachineVirtualCamera == null)
+                            continue;
+
+                        var behaviour = cinemachineVirtualCamera.GetCinemachineComponent(components[i].Stage) as MonoBehaviour;
+                        if (behaviour != null)
+                            behaviours.Add(behaviour);
+                    }
+
+                    var behaviourArray = behaviours.ToArray();
+                    if (behaviourArray.Length > 0 && behaviourArray[0] != null && (hasSameStageDataTypes[(int)components[i].Stage] || targets.Length == 1))
```

Comment:
Why the `behaviourArray[0] != null` check is needed? I don't quite see how would a null entry get in there

Reply:
Yep, leftovers from when I used to cast it to a different type. Will fix.